Fix named tensor printing

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25564 Fix named tensor printing**

There are a number of ops that get called while printing tensors
depending on how large the tensors are. This PR makes it so that before
we attempt to format tensor data for printing, we drop the names of the
tensor (if there are any). This is easier than supporting named tensors
for all of those ops (which should happen eventually).

Test Plan:
- new test [namedtensor ci]

Differential Revision: [D17158872](https://our.internmc.facebook.com/intern/diff/D17158872)

zou3519 committed:
Fix named tensor printing

There are a number of ops that get called while printing tensors
depending on how large the tensors are. This PR makes it so that before
we attempt to format tensor data for printing, we drop the names of the
tensor (if there are any). This is easier than supporting named tensors
for all of those ops (which should happen eventually).

Test Plan:
- new test [namedtensor ci]

nairbv commented on test/test_namedtensor.py:
```diff
@@ -183,6 +183,14 @@ def test_no_multiprocessing_support(self):
+            expected = "{}, names={})".format(repr(unnamed_tensor)[:-1], named_tensor.names)
+            self.assertEqual(repr(named_tensor), expected)
+
+        check_repr(torch.randn(128, 3, 64, 64, names=('N', 'C', 'H', 'W')))
```
I think the test in this case would have passed with or without your change to `_tensor_str`, what's the case where the names will now be missing from the repr?

zou3519 replied:
The test case crashes with `RuntimeError: cat: no named inference rule implemented.`. We use a lot of various ops to format the tensor's data into a form that can be printed. If the tensor's data is too large to fit on screen, it needs to be summarized, which is where the `torch.cat` call came from.

nairbv replied:
ah you mean to generate the truncated version of the tensor data? maybe we can make that comment below clearer, the meaning wasn't obvious to me.

zou3519 replied:
Done

zou3519 committed:
Update on "Fix named tensor printing"


There are a number of ops that get called while printing tensors
depending on how large the tensors are. This PR makes it so that before
we attempt to format tensor data for printing, we drop the names of the
tensor (if there are any). This is easier than supporting named tensors
for all of those ops (which should happen eventually).

Test Plan:
- new test [namedtensor ci]

Differential Revision: [D17158872](https://our.internmc.facebook.com/intern/diff/D17158872)
